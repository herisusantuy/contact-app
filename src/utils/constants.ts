export const testID = {
  swipeableWrapper: "card-contact-swipeable-wrapper",
  photoProfile: "card-contact-photo-profile",
  fullnameText: "card-contact-fullname-text",
  ageText: "card-contact-age-text",
  pressableView: "floating-button-pressable-view",
  plusIcon: "floating-button-plus-icon",
  textInput: "form-text-field",
  customModal: "custom-modal",
  cancelButton: "cancel-button",
  saveButton: "save-button",
};
