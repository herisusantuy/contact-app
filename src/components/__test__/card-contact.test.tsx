import { describe, expect, test } from "@jest/globals";
import CardContact from "../card-contact";
import { renderWithProviders } from "../../utils/test-utils";
import { testID } from "../../utils/constants";

describe("Testing Card ", () => {
  it("Should render Swipeable Wrapper, Photo Profile, Fullname Text and Age text", () => {
    const dummy = {
      firstName: "Bilbo",
      lastName: "Baggins",
      age: 111,
      photo:
        "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
      id: "cb41c300-f1c5-11ee-8e3f-71d14f52ed26",
    };

    const { getByTestId } = renderWithProviders(
      <CardContact contact={dummy} />
    );
    const swipeableWrapper = getByTestId(testID.swipeableWrapper);
    const photoProfile = getByTestId(testID.photoProfile);
    const fullnameText = getByTestId(testID.fullnameText);
    const ageText = getByTestId(testID.ageText);

    expect(swipeableWrapper).toBeTruthy();
    expect(photoProfile).toBeTruthy();
    expect(fullnameText).toBeTruthy();
    expect(ageText).toBeTruthy();
  });
});
