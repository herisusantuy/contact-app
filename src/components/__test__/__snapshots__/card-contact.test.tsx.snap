// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card contact Card contact render correctly.. 1`] = `
<View>
  <View
    collapsable={false}
    handlerTag={5}
    handlerType="PanGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      {
        "overflow": "hidden",
      }
    }
    testID="card-contact-swipeable-wrapper"
  >
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "flexDirection": "row-reverse",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateX": -10000,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minHeight": 50,
            "width": 100,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "space-between",
              "opacity": 0,
              "paddingHorizontal": 15,
              "width": 100,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            delayLongPress={600}
            enabled={true}
            exclusive={true}
            handlerTag={6}
            handlerType="NativeViewGestureHandler"
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            touchSoundDisabled={false}
          >
            <View
              accessible={true}
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "black",
                      "fontSize": 24,
                    },
                    undefined,
                    {
                      "fontFamily": "feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </RNGestureHandlerButton>
          <RNGestureHandlerButton
            collapsable={false}
            delayLongPress={600}
            enabled={true}
            exclusive={true}
            handlerTag={7}
            handlerType="NativeViewGestureHandler"
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            touchSoundDisabled={false}
          >
            <View
              accessible={true}
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "black",
                      "fontSize": 24,
                    },
                    undefined,
                    {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
      <View
        onLayout={[Function]}
      />
    </View>
    <View
      collapsable={false}
      handlerTag={8}
      handlerType="TapGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      pointerEvents="auto"
      style={
        {
          "transform": [
            {
              "translateX": -0,
            },
          ],
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 5,
            "minHeight": 50,
            "paddingHorizontal": 10,
            "paddingLeft": 5,
          }
        }
      >
        <Image
          height={50}
          resizeMode="cover"
          source={
            {
              "uri": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
            }
          }
          style={
            {
              "borderRadius": 50,
              "marginRight": 10,
            }
          }
          testID="card-contact-photo-profile"
          width={50}
        />
        <View
          style={
            {
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "600",
              }
            }
            testID="card-contact-fullname-text"
          >
            Bilbo Baggins
          </Text>
          <Text
            style={
              {
                "fontSize": 14,
                "fontWeight": "normal",
              }
            }
            testID="card-contact-age-text"
          >
            Age: 
            111
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
